test/karma.conf.js

```diff
@@ -3,12 +3,31 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
+'use strict';
+
+const isDocker = require('is-docker');
+
 // Karma configuration
 // http://karma-runner.github.io/0.12/config/configuration-file.html
 
-'use strict';
 module.exports = function(config) {
+  // see https://github.com/docker/for-linux/issues/496
+  const disableChromeSandbox = isDocker && !process.env.TRAVIS;
+  if (disableChromeSandbox) {
+    console.log('!! Disabling Chrome sandbox to support un-privileged Docker !!');
+  }
+
   config.set({
+    customLaunchers: {
+      ChromeDocker: {
+        base: 'ChromeHeadless',
```

Comment:
If you change this to `ChromiumHeadless` it might "just work" since Chromium is installed.

Reply:
In c772233, I added a check to see if `chromium` or `chromium-browser` is in `PATH` and then use `ChromiumHeadless` instead of `ChromeHeadless`. Unfortunately, it seems the check did not pass, because the build is still using Chrome (not Chromium):

https://cis-jenkins.swg-devops.com/job/nb/job/loopback%7Emaster/label=x64%20&&%20linux%20&&%20nvm,nodeVersion=10/300/console

Reply:
Below is the Dockerfile I am using to test my setup, it's based on our build images.

```Dockerfile
###
### Copied from Jenkins base Dockerfile
###

FROM ubuntu:xenial

# Declared first so that during build, nothing ever gets accidentally persisted
VOLUME /tmp

# Simple wrapper for apt-get install that does an update before install and
# cleanup after install to ensure that the resulting layer doesn't contain
# anything extra
COPY docker-friendly-apt-install.sh /bin/docker-friendly-apt-install

# Ensure that package don't give us any trouble when we install them
RUN echo 'DPkg::options { "--force-confdef"; "--force-confnew"; }' \
    > /etc/apt/apt.conf.d/60local

# Use bash by default instead of dash
RUN echo "dash dash/sh boolean false" | debconf-set-selections \
 && DEBIAN_FRONTEND=noninteractive dpkg-reconfigure dash

# Set our locale to use UTF-8 by default
RUN docker-friendly-apt-install locales \
 && locale-gen en_US.UTF-8 \
 && update-locale LANG=en_US.UTF-8 LANGUAGE=en_US:en LC_ALL=en_US.UTF-8
ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'

# ssh and some additional packages that are sufficiently common/popular that
# they are probably wanted/needed in all images.
RUN docker-friendly-apt-install \
      ca-certificates \
      apt-transport-https \
      wget \
      curl \
      git \
      openssh-server \
      zip \
      unzip \
      bzip2 \
      xz-utils \
      bash \
      rsync \
      jq

###
### Install Chromium - see xvfb/Dockerfile
###

# For running protractor/selenium based tests
# libgconf-2-4 is a dependency of the selenium chromedriver
RUN docker-friendly-apt-install xvfb chromium-chromedriver libgconf-2-4

###
### Setup Node environment
###

# Install nvm
RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | bash

RUN source ~/.nvm/nvm.sh && nvm install 12

###
### Run our tests
###

ENV CI=true
ENV JENKINS_HOME=/tmp

COPY "." "."

CMD ["/bin/bash", "-c", "source ~/.nvm/nvm.sh && nvm run 12 ./node_modules/.bin/grunt karma:unit-once"]
```

I am running the following command to build & run the tests:

```
docker build -t loopback-core-tests . && docker run -it loopback-core-tests
```

When I run this on my machine, Chromium is correctly picked and all tests pass.

@rmg could you please help me to investigate why my tests are failing on cis-jenkins?